pymatbridge/messenger/make.py

```diff
@@ -0,0 +1,207 @@
+                    state = state_whitespace
+            else:
+                arg = arg + c
+                state = state_basic
+        elif state == state_esc:
+            arg = arg + c
+            state = state_basic
+        elif state == state_singlequote:
+            if c == r"'":
+                state = state_basic
+            else:
+                arg = arg + c
+        elif state == state_doublequote:
+            if c == r'"':
+                state = state_basic
+            else:
+                arg = arg + c
+
+    if arg != '':
+        arg_list.append(arg)
+    return arg_list
+
+def get_messenger_dir():
+    host, is_64bit = platform.system(), platform.machine().endswith('64')
+    ostype = {
+        'Darwin':  'mexmaci64',
+        'Linux':   'mexa64',
+        'Windows': 'mexw64',
+    }
+    if not is_64bit and host == 'Windows':
+        raise ValueError("pymatbridge does not support 32-bit Windows")
+
+    return ostype[host] if is_64bit else 'mexw32'
+
+
+def get_config(host, config='config.ini'):
+
+    cfg = ConfigParser()
+    cfg.read(config)
+
+    return dict(cfg.items(host))
+
+
+def do_build(make_cmd, messenger_exe):
+
+    messenger_dir = get_messenger_dir()
+    use_shell     = sys.platform.startswith('win32')
+
+    print('Building %s...' % messenger_exe)
+    print(make_cmd)
+
+    subprocess.check_output(make_cmd, shell=use_shell)
+
+    messenger_loc = os.path.join(messenger_dir, messenger_exe)
+
+    shutil.move(messenger_exe, os.path.join('messenger', messenger_loc))
+
+    if os.path.isfile('messenger.o'):
+        os.remove('messenger.o')
+
+
+def build_octave(static=False):
+
+    paths = "-L%(octave_lib)s -I%(octave_inc)s -L%(zmq_lib)s -I%(zmq_inc)s"
+    paths = paths % get_config(platform.system())
+    make_cmd = "mkoctfile --mex %s -lzmq messenger/src/messenger.c" % paths
+
+    do_build(split_command_line(make_cmd), 'messenger.mex')
+
+
+def build_matlab(static=False, messenger='messenger.c'):
+    """
+    Build the messenger mex for MATLAB
+
+    Parameters
+    ----------
+    static : bool
+        Determines if the zmq library has been statically linked.
+        If so, it will append the command line option -DZMQ_STATIC
+        when compiling the mex so it matches libzmq.
+    """
+    matlab_bin, cfg = get_matlab_bin(), get_config(platform.system())
+
+    use_shell  = sys.platform.startswith('win32')
+    extcmd     = split_command_line(os.path.join(matlab_bin, "mexext"))
+    extension  = subprocess.check_output(extcmd, shell=use_shell)
+    extension  = extension.decode('utf-8').rstrip()
+
+    # Build the mex file
+    mex = os.path.join(matlab_bin, "mex")
+    paths    = "-L'%(zmq_lib)s' -I'%(zmq_inc)s'" % cfg
+    make_cmd = '%s -O %s -lzmq %s' % (mex, paths, messenger)
+
+    if static:
+        make_cmd += ' -DZMQ_STATIC'
+
+    print(make_cmd)
+    subprocess.check_output(split_command_line(make_cmd), shell=use_shell)
+
+
+def get_matlab_bin(config='config.ini'):
```

Comment:
Please replace this with the functionality in the current master branch's `which_matlab`, using `pexpect`, etc.